### PR TITLE
ClientBuilder: Allow message delays of 0 ms.

### PR DESCRIPTION
**Pull request by Zarthus**
This seemed like an arbitrary restriction to me.

While I haven't tested if this makes KICL go bonkers (and I'm hoping travis catches it if it does), this should have no negative repercussions and in theory would be an arbitrary restriction.

Primarily useful for when you want to make dynamic configurations that implement KICL, the chances someone will set this to "0" over "1" is quite high, and it makes sense to allow 0 for that.

### CODE REVIEW

File: src/main/java/org/kitteh/irc/client/library/implementation/ClientBuilder.java [hunk 1]
```diff
@@ -162,8 +162,15 @@ public ClientBuilder secureTrustManagerFactory(@Nullable TrustManagerFactory fac
     @Nonnull
     @Override
     public ClientBuilder messageDelay(int delay) {
-        Sanity.truthiness(delay > 0, "Delay must be at least 1");
-        this.config.set(Config.MESSAGE_DELAY, delay);
+        Sanity.truthiness(delay >= 0, "Delay must be a positive integer");
```
Comment (lol768) [inline src/main/java/org/kitteh/irc/client/library/implementation/ClientBuilder.java]: >positive integer



Consider rewording error.

**Comment by lol768** [comment]
> Due to limitations with ScheduledExecutorService, we need a delay of at least 1 ms.

This seems like a hack. Why does `NettyManager#schedule` **need** to use `ScheduledExecutorService` if it detects the delay is 0?

**Comment by mbax** [comment]
Thanks for the push I needed to rewrite all the things! 

https://github.com/KittehOrg/KittehIRCClientLib/commit/4278fc9d98d4cf1a93f32a105ee841038abc4610

**Comment by Zarthus** [comment]
Glad you liked at least 100% of my original code.

**Pull request closed without merging**